catalog/from-k8s/resource.go

```diff
@@ -47,6 +47,11 @@ type ServiceResource struct {
 	// Setting this to false will ignore ClusterIP services during the sync.
 	ClusterIPSync bool
 
+	// NodeExternalIPSync set to true (the default) syncs NodePort services
+	// using the node's external ip address. When false, the node's internal
+	// ip address will be used instead.
+	NodeExternalIPSync bool
```

Comment:
We don't have to, but I wonder if we should change this to take an enum of values so we can support the various ways people do NodePort syncs. I'm not sure if its necessary, but don't want to end up with boolean-hell one day with a bunch of modes. 😄

Reply:
That's a good thought, and allows me to address the case where someone might want the ExternalIP if it's available, but wants to fall back to the InternalIP case.